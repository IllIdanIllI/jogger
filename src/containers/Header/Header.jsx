import React, { useState, useEffect, useRef } from 'react';
import { Link, useLocation } from 'react-router-dom';
import { useSelector } from 'react-redux';
import menu from '../../images/menu/menu.png';
import { ReactComponent as Cross } from '../../images/cancel/cancel.svg';
import { ReactComponent as Logo } from '../../images/logo/logo.svg';
import { ReactComponent as Filter } from '../../images/filter/filter.svg';
import './Header.scss';
import { JOGS_URL, INFO_URL, CONTACT_URL } from '../../constants/constants';
import { toggleFilter } from '../../actions/jogsActions';
import { useActions } from '../../custom/customHooks';

const headerItems = [
    { name: 'Jogs', path: JOGS_URL },
    { name: 'Info', path: INFO_URL },
    { name: 'Contact Us', path: CONTACT_URL },
];

const Header = () => {
    const isAuthenticated = useSelector(
        (state) => state.authentication.isAuthenticated
    );
    const isFilterActive = useSelector(
        (state) => state.jogStore.isFilterActive
    );

    const [checked, setChecked] = useState(false);

    const [onToggleFilter] = useActions([toggleFilter], []);

    const location = useLocation();

    useEffect(() => {
        if (location.pathname !== JOGS_URL && isFilterActive) {
            onToggleFilter(isFilterActive);
        }
    }, [location.pathname]);

    return (
        <header>
            <Link to="/">
                <Logo
                    className={`header__logo ${
                        checked ? 'header__logo_green' : ''
                    }`}
                />
            </Link>

            {isAuthenticated && (
                <nav>
                    <input
                        type="checkbox"
                        checked={checked}
                        id="switcher"
                        readOnly
                    />
                    <label htmlFor="switcher" className="show-menu-btn">
                        <img
                            src={menu}
                            alt="Menu"
                            onClick={() => setChecked(true)}
                        />
                    </label>
                    <div className={`header__items-filter`}>
                        <Filter
                            onClick={() =>
                                location.pathname === JOGS_URL &&
                                onToggleFilter(isFilterActive)
                            }
                            className={`header__items-filter-instance${
                                isFilterActive ? '' : '_inactive'
                            }`}
                        />
                    </div>
                    <ul className="header__items">
                        {headerItems.map((item) => (
                            <Link
                                className={`header__items-instance`}
                                key={item.name}
                                to={item.path}
                                onClick={() => {
                                    if (checked) {
                                        setChecked(false);
                                    }
                                }}
                            >
                                {item.name}
                            </Link>
                        ))}
                        <label htmlFor="switcher" className="hide-menu-btn">
                            <Cross onClick={() => setChecked(false)} />
                        </label>
                    </ul>
                </nav>
            )}
        </header>
    );
};

export default Header;
