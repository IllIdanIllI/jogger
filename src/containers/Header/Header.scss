@import '../../style/colors';

@mixin a-teg-decoration {
    cursor: pointer;
    text-transform: uppercase;
    font-size: 14px;
    font-weight: bold;
    color: white;
    text-decoration: none;
}

header {
    width: 100%;
    height: 116px;
    position: fixed;
    z-index: 10;
    background-color: $apple-green;
    .header {
        &__logo {
            margin-top: 28px;
            margin-left: 37px;
            float: left;
            z-index: 25;
            &_green {
                position: relative;
                g {
                    fill: $apple-green;
                }
            }
        }

        &__items {
            float: right;
            line-height: 116px;
            margin: 0;

            &-instance {
                color: #fff;
                position: relative;
                text-transform: uppercase;
                text-decoration: none;
                padding: 0 20px;
                transition: 0.4s;

                @include a-teg-decoration();
                &:hover:after {
                    position: absolute;
                    content: '';
                    height: 2px;
                    bottom: -5px;
                    margin: 0 auto;
                    left: 0;
                    right: 0;
                    width: 80%;
                    background: white;
                }
            }

            &-filter {
                position: relative;
                top: 37px;
                margin-right: 20px;
                float: right;
                cursor: pointer;
                &-instance {
                    &_inactive {
                        cursor: pointer;
                        g {
                            path {
                                fill: #ffffff;
                            }
                            rect {
                                stroke-width: 0;
                            }
                        }
                    }
                }
            }
        }
    }
    .show-menu-btn,
    .hide-menu-btn {
        transition: 0.4s;
        cursor: pointer;
        display: none;
    }

    .show-menu-btn {
        float: right;
    }

    #switcher {
        position: absolute;
        visibility: hidden;
        z-index: -10;
    }
}

@media screen and (max-width: 800px) {
    header {
        .header {
            &__items {
                position: fixed;
                width: 100%;
                height: 100vh;
                background-color: white;
                right: -100%;
                top: 0;
                text-align: center;
                padding: 80px 0;
                line-height: normal;
                transition: 0.7s;

                &-instance {
                    color: $dark-grey;
                    font-size: 20pt;
                    display: block;
                    padding: 20px;
                    &:hover {
                        color: $apple-green;
                    }
                }

                &-filter {
                    right: 5vw;
                }
            }
        }

        .show-menu-btn {
            display: block;
            margin-top: 45px;
            margin-right: 36px;
        }

        .hide-menu-btn {
            display: block;
            position: absolute;
            top: 40px;
            right: 40px;
            svg path {
                fill: $text-grey;
            }
        }

        #switcher:checked ~ .header__items {
            right: 0;
        }
    }
}
