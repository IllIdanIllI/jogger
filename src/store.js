import { createStore, combineReducers, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import authentication from './reducers/authenticationReducer';
import jogStore from './reducers/jogsReducer';
const rootReducer = combineReducers({
    authentication,
    jogStore,
});

const middleware = [thunk];

const store = createStore(
    rootReducer,
    composeWithDevTools(applyMiddleware(...middleware))
);
export default store;
