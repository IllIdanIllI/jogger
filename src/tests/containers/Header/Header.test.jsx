import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../store';
import Header from '../../../containers/Header';
import renderer from 'react-test-renderer';

test('Header test', () => {
    const component = renderer.create(
        <Provider store={store}>
            <Router>
                <Header />
            </Router>
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
