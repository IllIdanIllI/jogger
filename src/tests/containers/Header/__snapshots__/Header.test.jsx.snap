// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test 1`] = `
<header>
  <a
    href="/"
    onClick={[Function]}
  >
    <svg
      className="header__logo "
    >
      logo.svg
    </svg>
  </a>
</header>
`;
